File: matiasmasso/Mat.NET/DT2/DTOEmail.vb
Public Class DTOEmail
    Inherits DTOBaseGuid


    Property EmailAddress As String
    Property BadMail As BadMailErrs
    Property Obs As String
    Property Privat As Boolean
    Property Obsoleto As Boolean

    Public Enum BadMailErrs
        None
        FullMailBox
        ServerNotFound
        UserNotFound
        BlackList
        Altres
    End Enum

    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
    End Sub

    Shared Function MailAddressCollection(oEmails As List(Of DTOEmail)) As System.Net.Mail.MailAddressCollection
        Dim retval As New System.Net.Mail.MailAddressCollection
        For Each oEmail As DTOEmail In oEmails
            retval.Add(oEmail.EmailAddress)
        Next
        Return retval
    End Function

    Shared Function Recipients(oEmails As List(Of DTOEmail)) As String
        Dim sb As New Text.StringBuilder
        For Each oEmail As DTOEmail In oEmails
            If oEmail.UnEquals(oEmails.First) Then
                sb.Append("; ")
            End If
            sb.Append(oEmail.EmailAddress)
        Next
        Dim retval As String = sb.ToString
        Return retval
    End Function
End Class

File: matiasmasso/Mat.NET/DTO/DTOEmail.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace DTO
{
    public class DTOEmail : DTOBaseGuid
    {
        public string EmailAddress { get; set; }
        public DTOCod BadMail { get; set; }
        public string Obs { get; set; }
        public bool Privat { get; set; }
        public bool Obsoleto { get; set; }

        public enum BadMailErrs
        {
            None,
            FullMailBox,
            ServerNotFound,
            UserNotFound,
            BlackList,
            Altres
        }

        public DTOEmail() : base()
        {
        }

        public DTOEmail(Guid oGuid) : base(oGuid)
        {
        }

        public static System.Net.Mail.MailAddressCollection MailAddressCollection(List<DTOEmail> oEmails)
        {
            System.Net.Mail.MailAddressCollection retval = new System.Net.Mail.MailAddressCollection();
            foreach (DTOEmail oEmail in oEmails)
                retval.Add(oEmail.EmailAddress);
            return retval;
        }

        public static string Recipients(List<DTOEmail> oEmails)
        {
            System.Text.StringBuilder sb = new System.Text.StringBuilder();
            foreach (DTOEmail oEmail in oEmails)
            {
                if (oEmail.UnEquals(oEmails.First()))
                    sb.Append("; ");
                sb.Append(oEmail.EmailAddress);
            }
            string retval = sb.ToString();
            return retval;
        }
    }
}
